iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/expected.inl

```diff
@@ -29,17 +29,20 @@ template <typename... T>
 struct IsOptional : std::false_type
 {
 };
+// AXIVION Next Construct AutosarC++19_03-A14.7.2 : We are extending std, also, it is allowed as
```

Comment:
Actually no, see this thing here:
```cxx
template<>
class optional;

template <typename T>
struct is_optional : std::false_type {};

template <typename T>
struct is_optional<optional<T>> : std::true_type {};
```

No dependencies at all and the compile time should also not increase.

On a side note, for me `type_traits` was always the place were all type deduction thingies go and with this in mind it seemed consistent.

Reply:
@elfenpiff @elBoberido I would prefer to refrain from moving all `is_foo` type traits to the generic `type_traits.hpp`. I know that a forward declaration is sufficient here, but it still feels weird. The STL also has only fundamental types in it's `<type_traits>` header. On #1791 I moved `is_iox_string` to a separate header file and discussed on #1785 that `is_span` should also not go into `type_traits.hpp`.

cc @MatthiasKillat

Reply:
I agree. And to my knowledge the STL does this in the header of the individual classes as well, but I need to check. This is reasonable, as otherwise it might increase the scope of `type_traits.hpp` too much.